protocol/communities/community_description_encryption.go

```diff
@@ -59,15 +60,31 @@ func encryptDescription(encryptor DescriptionEncryptor, community *Community, de
 	return nil
 }
 
+type CommunityPrivateDataFailedToDecrypt struct {
+	GroupID []byte
+	KeyID   []byte
+}
+
 // Decrypts members and chats
-func decryptDescription(encryptor DescriptionEncryptor, description *protobuf.CommunityDescription, logger *zap.Logger) error {
+func decryptDescription(id types.HexBytes, encryptor DescriptionEncryptor, description *protobuf.CommunityDescription, logger *zap.Logger) ([]*CommunityPrivateDataFailedToDecrypt, error) {
+	if len(description.PrivateData) == 0 {
+		return nil, nil
+	}
+
+	var failedToDecrypt []*CommunityPrivateDataFailedToDecrypt
+
 	for keyIDSeqNo, encryptedDescription := range description.PrivateData {
-		decryptedDescription, err := encryptor.decryptCommunityDescription(keyIDSeqNo, encryptedDescription)
+		decryptedDescriptionResponse, err := encryptor.decryptCommunityDescription(keyIDSeqNo, encryptedDescription)
+		if decryptedDescriptionResponse != nil && !decryptedDescriptionResponse.Decrypted {
+			failedToDecrypt = append(failedToDecrypt, &CommunityPrivateDataFailedToDecrypt{GroupID: id, KeyID: decryptedDescriptionResponse.KeyID})
```

Comment:
This loop also iterates thorugh encrypted channels, GroupID is not alwas communityID (can be communityID+channelID).

Reply:
There is a complication because we cannot retrieve the `GroupID` from the `keyID`, which prevents us from `SaveHashRatchetMessage`. Additionally, storing the hash ratchet message for future use implies processing the same message more than once, which could lead to further issues.

The cleanest solution to me would be to invoke `decryptDescription` on already processed community once we receive a new key for it.

Reply:
>This loop also iterates thorugh encrypted channels, GroupID is not alwas communityID (can be communityID+channelID).

Ah, this I didn't know, let me check if I can replicate a failure in a test.

> Additionally, storing the hash ratchet message for future use implies processing the same message more than once, which could lead to further issues.

This I am not sure I understand fully, could you elaborate? (we intentionally reprocess the same message i.e the community description), but I think I might misunderstand you

Reply:
> This I am not sure I understand fully, could you elaborate? (we intentionally reprocess the same message i.e the community description), but I think I might misunderstand you

Instead processing the same message and making things like https://github.com/status-im/status-go/pull/4129 impossible/difficult to implement we could introduce `HandleHashRatchetKeys` on a messenger level that will:
```
communityID := getCommunityID(groupID)
...
community := m.manager.GetByID(communityID)
...
response := m.manager.DecryptCommunity(community)
...
return response
```